sound/soc/sof/intel/hda-ipc.c

```diff
@@ -244,6 +244,9 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 
 	spin_lock(&sdev->hw_lock);
 
+	if (!sdev->ipc_enabled)
+		goto out;
```

Comment:
this is really interesting. In the SoundWire work, we also get interrupts on WHL before they are even enabled. There is something really fishy here, it's as if the interrupt masking doesn't fully work. We should check with hardware folks here.